lib/modules/apostrophe-login/index.js

```diff
@@ -305,12 +306,51 @@ module.exports = {
       return self.apos.users.safe.findOne({
         _id: user._id
       }).then(function(userSafe) {
-        return done(null, userSafe && userSafe.totp && userSafe.totp.key, userSafe && userSafe.totp && userSafe.totp.period);
+        return self.disableIfInactive(
+          userSafe,
+          done,
+          {
+            key: userSafe && userSafe.totp && userSafe.totp.key,
+            period: userSafe && userSafe.totp && userSafe.totp.period
+          }
+        );
       }).catch(function(err) {
         return done(err);
       });
     };
 
+    self.disableIfInactive = function(user, callback, totp) {
```

Comment:
This function now has a muddled purpose. It handles disabling if inactive, but it also handles returning the totp information, which is messy.

It should have a single purpose, and the piece of code that cares about both tasks should be responsible for handling the totp information when there is no error from it.

Reply:
Ok @boutell 
Here is the new proposal